enhancements/network/user-defined-network-for-virt-workloads.md

```diff
@@ -0,0 +1,595 @@
+
+Passt relies on a user space program running on the pod namespace that maps
+traffic from the guest to the respective socket in the host; it currently is
+**not** migrated to the destination pod during migration, which causes the
+established TCP connections to be severed when the VM is migrated.
+
+This requires an enhancement on passt.
+
+TODO: link the ticket to get an enhancement
+
+### Persisting VM IP addresses during the migration
+
+OpenShift already features the ability of providing persistent IP addresses
+for layer2 **secondary** networks. It relies on the
+[IPAMClaim CRD](https://github.com/k8snetworkplumbingwg/ipamclaims/blob/48c5a915da3b67f464a4e52fa50dbb3ef3547dcd/pkg/crd/ipamclaims/v1alpha1/types.go#L23)
+to "tie" the allocation of the IP address to the VM.
+In short, KubeVirt (or a component on behalf
+of KubeVirt) creates an `IPAMClaim` resource for each interface in the VM the
+user wants to be persistent, and KubeVirt instructs the CNI plugin of which
+claim to use to persist the IP for the VM using an attribute defined in the
+k8snetworkplumbingwg network-selection-elements (the
+`k8s.v1.cni.cncf.io/networks` annotation). This protocol is described in depth
+in
+[the persistent IPs for virt workloads enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/network/persistent-ips-sdn-secondary-networks.md#creating-a-virtual-machine).
+
+We want to re-use this mechanism to implement persistent IPs for UDN networks;
+meaning KubeVirt (or a component on behalf of it) will create the `IPAMClaim`
+and will instruct OVN-Kubernetes of which claim to use. Since for primary UDNs
+we do **not** define the network-selection-elements, we need to use a new
+annotation to pass this information along.
+
+The proposed annotation value is `k8s.ovn.org/ovn-udn-ipamclaim-reference`.
+
+These persistent IPs will be cleaned up by the Kubernetes garbage collector
+once the VM to which they belong is removed.
+
+All the other
+[work required in OVN-Kubernetes](https://github.com/trozet/enhancements/blob/941f5c6391830d5e4a94e65d742acbcaf9b8eda9/enhancements/network/user-defined-network-segmentation.md#pod-egress)
+to support live-migration was already implemented as part of the epic
+implementing UDN.
+
+### Allow user to configure the VMs interface desired IP address
+
+We will also rely on the `IPAMClaim` CRD to provide the VM owner a way to
+specify the IP addresses for the UDN interface of their VMs. This way, we
+do not require to update the KubeVirt API, which is something we have faced
+extreme resistance in the past.
+
+Keep in mind that up to now the IPAMClaims would be created by KubeVirt
+directly; we need to allow the VM user to create them, specifying in its `spec`
+the desired IP addresses for the VM, and when creating the VM, point it to the
+respective `IPAMClaim` using an annotation (on the VM).
+
+KubeVirt (or a component on behalf of it) will see the annotation on the VM,
```

Comment:
here it is even an action item, not just something that already exists
the ipam-ext should do it right ? i think it would be better to be precise here, and not mention kubevirt ?

Reply:
Yes, it would that component. I just don't understand what anyone gets from having to know there's an extra moving piece. 

What is important is that it is something that understands what a VM is.